lib/text/simple_text_displayer.js

```diff
@@ -121,16 +121,29 @@ shaka.text.SimpleTextDisplayer = class {
     // We don't want to modify the array or objects passed in, since we don't
     // technically own them.  So we build a new array and replace certain items
     // in it if they need to be flattened.
-    const flattenedCues = cues.map((cue) => {
-      if (cue.nestedCues.length) {
-        const flatCue = cue.clone();
-        flatCue.nestedCues = [];
-        flatCue.payload = flattenPayload(cue);
-        return flatCue;
-      } else {
-        return cue;
+    // We also don't want to flatten the text payloads starting at a container
+    // element; otherwise, for containers encapsulating multiple caption lines,
+    // the lines would merge into a single cue. This is undesirable when a
+    // subset of the captions are outside of the append time window. To fix
+    // this, we only call flattenPayload() starting at elements marked as
+    // isContainer = false.
+    let flattenedCues = [];
+    const getCuesToFlatten = (cues) => {
+      for (const cue of cues) {
+        if (cue.isContainer) {
+          flattenedCues =
```

Comment:
I have a problem with the data flow here.  flattenedCues is set to the result of getCuesToFlatten, which itself modifies flattenedCues in multiple ways and then returns flattenedCues.  This is very hard to follow.

If getCuesToFlatten is going to be recursive, you shouldn't modify an effectively "global" variable (not truly global, but captured by this closure, so the effect is similar).